src/cart/crt_context.c

```diff
@@ -851,10 +853,29 @@ crt_context_timeout_check(struct crt_context *crt_ctx)
 	struct d_binheap_node		*bh_node;
 	d_list_t			 timeout_list;
 	uint64_t			 ts_now;
-	int				 count = 0;
 
 	D_ASSERT(crt_ctx != NULL);
 
+	if (crt_initialized() && crt_is_service() && crt_gdata.cg_swim_inited) {
+		struct crt_grp_priv	*gp = crt_gdata.cg_grp->gg_primary_grp;
+		struct crt_swim_membs	*csm = &gp->gp_membs_swim;
+
+		if (crt_ctx->cc_idx == csm->csm_crt_ctx_idx &&
+		    crt_ctx->cc_last_unpack_hlc != 0) {
+			uint64_t delay = crt_hlc2msec(crt_hlc_get() -
+						   crt_ctx->cc_last_unpack_hlc);
```

Comment:
the logical part may be lost in noise when dividing.  There are 18 bits of logical time (256K) and it gets divided by 1 million so at most it would be off by half a millisecond?   Unless the question is whether it's safe to use hlc at all for this calculation?   It seems like it would also be reasonable since we won't start servers if clock drift is too high.

Reply:
It's okay to subtract HLC timestamps. We just need to keep in mind that the precision is around 16 us, rather than 1/16 ns.

Reply:
Totally agree with LiWei, this should be safe because of we most likely will loose less significant bits when will get milliseconds. Also less significant bits in HLC is just incremental counter and in worst case after substraction we just got result which decreased by 16us (less accurate). This is only consequences I see with HLC substraction.
Finally I use crt_hlc2msec() which correctly converts time encoded into HLC into milliseconds.